### DISCUSSION TITLE
Make PNG a module

### DISCUSSION THREAD

**Discussion by DeeJayLSP** [godotengine/godot-proposals#4471] [Editor]
I don't know if this is possible. With the amazing support Godot has for WebP, I don't see why someone would use both.

**Comment by Mickeon** [comment]
Could this be elaborated? I don't see why, currently. PNG is so widely common, it is to be expected most programs should natively support it.

**Reply by DeeJayLSP** [reply]
WebP is a module, JPG is a module, BMP is a module, TGA is a module.

If people can have an option to remove support for those, then they should have an option to remove PNG support too.

Currently, even if you don't use PNGs, custom builds must have PNG support enabled, no matter what (unless you modify the source code, which isn't an easy process for most people).

Even GDScript is a module.

**Reply by Mickeon** [reply]
Aaah, I understand now. Although, unfortunately, PNG may be a bit too integrated within the Engine, so it may not be nearly as easy as just allowing it away.

**Comment by Calinou** [comment]
Superseded by https://github.com/godotengine/godot-proposals/issues/4475.
